lib/spack/docs/module_file_support.rst

```diff
@@ -0,0 +1,664 @@
+``spack unload <spec>``    ``spack unuse <spec>``
+=========================  ==========================
+
+And you can use the same shortened names you use everywhere else in
+Spack.  For example, this will add the ``mpich`` package built with
+``gcc`` to your path:
+
+.. code-block:: console
+
+   $ spack install mpich %gcc@4.4.7
+
+   # ... wait for install ...
+
+   $ spack use mpich %gcc@4.4.7
+   Prepending: mpich@3.0.4%gcc@4.4.7 (ok)
+   $ which mpicc
+   ~/src/spack/opt/linux-debian7-x86_64/gcc@4.4.7/mpich@3.0.4/bin/mpicc
+
+Or, similarly with modules, you could type:
+
+.. code-block:: console
+
+   $ spack load mpich %gcc@4.4.7
+
+These commands will add appropriate directories to your ``PATH``,
+``MANPATH``, ``CPATH``, and ``LD_LIBRARY_PATH``.  When you no longer
+want to use a package, you can type unload or unuse similarly:
+
+.. code-block:: console
+
+   $ spack unload mpich %gcc@4.4.7  # modules
+   $ spack unuse  mpich %gcc@4.4.7  # dotkit
+
+.. note::
+
+   These ``use``, ``unuse``, ``load``, and ``unload`` subcommands are
+   only available if you have enabled Spack's shell support *and* you
+   have dotkit or modules installed on your machine.
+
+----------------------
+Ambiguous module names
+----------------------
+
+If a spec used with load/unload or use/unuse is ambiguous (i.e. more
+than one installed package matches it), then Spack will warn you:
+
+.. code-block:: console
+
+   $ spack load libelf
+   ==> Error: Multiple matches for spec libelf.  Choose one:
+   libelf@0.8.13%gcc@4.4.7 arch=linux-debian7-x86_64
+   libelf@0.8.13%intel@15.0.0 arch=linux-debian7-x86_64
+
+You can either type the ``spack load`` command again with a fully
+qualified argument, or you can add just enough extra constraints to
+identify one package.  For example, above, the key differentiator is
+that one ``libelf`` is built with the Intel compiler, while the other
+used ``gcc``.  You could therefore just type:
+
+.. code-block:: console
+
+   $ spack load libelf %intel
+
+To identify just the one built with the Intel compiler.
+
+=======================
+Module files generation
+=======================
+
+Module files are generated by post-install hooks after the successful
+installation of a package. They are placed in the following directories
+under the Spack root:
+
+  +--------------------------------+------------------------------------+
+  |                                |  **Module files root**             |
+  +================================+====================================+
+  |  **Environment Modules**       |  share/spack/modules               |
+  +--------------------------------+------------------------------------+
+  |  **Dotkit**                    |   share/spack/dotkit               |
+  +--------------------------------+------------------------------------+
+  |  **LMod Hierarchical Modules** |   share/spack/lmod                 |
+  +--------------------------------+------------------------------------+
```

Comment:
This table is a little confusing. While serving a similar purpose, Dotkit files are not modules in this context. Also, LMod will work just fine with the modules installed in `share/spack/modules`. Likewise, modules written in Lua could also live in `share/spack/modules`, even along with the TCL(/C) environment module files. I get the implementation reason for writing out the hierarchical modules in Lua because that ensures that Lmod is used with those.

Maybe change as such

- 'Module files root' to 'Module/Dotkit files root'
- 'Environment Modules' to 'Environment/LMod Modules'